packages/myst-compat/src/downgrade/index.ts

```diff
@@ -0,0 +1,17 @@
+import type { IFile } from '../types/index.js';
```

Comment:
The idea here is that we accept unvalidated inputs, and enforce strict types inside the upgrade/downgrade routines